diff -purN a/primus_vk_diag.cpp b/primus_vk_diag.cpp
--- a/primus_vk_diag.cpp	2018-10-23 08:14:49.000000000 +0300
+++ b/primus_vk_diag.cpp	2018-10-23 18:37:56.911125386 +0300
@@ -180,7 +180,7 @@ void GLContext::drawSample(){
   gl->ptr_glClear (GL_COLOR_BUFFER_BIT);
   gl->ptr_glXSwapBuffers (data.display, data.win);
 
-  auto toSleep = timespec{.tv_nsec=200000000};
+  timespec toSleep = {0,200000000};
   nanosleep(&toSleep, nullptr);
 
   gl->ptr_glClearColor (1, 0.5, 0, 1);
